### ISSUE TITLE
Move searchbar to the titlebar

### ISSUE THREAD

**Issue by tannisroot** [lutris/lutris#2615]
Rough mockup made in Gimp (the search bar only appears when search button is pressed): 
![Screenshot from 2020-01-23 19-12-19](https://user-images.githubusercontent.com/10602045/73232365-0bf82b00-4194-11ea-97ff-fed87f7fadd0.png)
Many other Gnome apps already do this:
![Screenshot from 2020-01-23 19-10-17](https://user-images.githubusercontent.com/10602045/73232505-86c14600-4194-11ea-8268-29d993fbcdf6.png)

**Comment by luziferius** [comment]
When implementing this, make sure that there remains a fallback alternative. Not all window managers support this.

**Comment by tannisroot** [comment]
Which window managers don't support this?

**Comment by strycore** [comment]
I don't think there is one.
Worst case, the client side decoration adopt the Gtk window decorations and looks a bit out of place.
But I don't think there is a window manager that doesn't show CSD, we've been using those for a while and we would have heard about it if some WM didn't support those.

Did just means we don't need any fallback.

**Comment by luziferius** [comment]
I know quite some people personally who use XMonad and similar, and I wasn’t sure that it works in all cases.
And I had bad experience in a similar case with KWin in the past, where buttons in the window decorations are moved into the window and then won’t work: https://github.com/HandBrake/HandBrake/issues/2405

I just wanted to ping you to save trouble in the long run.

**Comment by strycore** [comment]
I believe the about dialog close button issue is different. I ran into this issue or something very similar at least with Lutris and that has been fixed since.
And I don't think that xmonad would behave any differently than any other WM that isn't part of Gnome.

The important thing to note is that we already use CSD and we have been for a while. If there were any kind of issue with any WM, we would know by now. This ticket is just about moving more widgets into the title bar. This is in no way a big change in our UI.
